src/main/java/com/hazelcast/remotecontroller/CloudManager.java

```diff
@@ -0,0 +1,428 @@
+                            "  \"planName\": \"%s\"," +
+                            "  \"hazelcastVersion\": \"%s\"," +
+                            "  \"tlsEnabled\": %b" +
+                            "}",
+                    clusterName,
+                    clusterTypeId,
+                    cloudProviderId,
+                    cloudProviderRegionId,
+                    clusterPlan,
+                    hazelcastVersion,
+                    isTlsEnabled);
+
+            LOG.info(String.format("Request query: %s", jsonString));
+            try (Response response = sendPostRequest("/cluster", jsonString)){
+                ResponseBody responseBody = response.body();
+                if (responseBody == null) {
+                    throw new CloudException(String.format("Response body is null while creating a dev mode cluster: %s", response));
+                }
+                String responseString = responseBody.string();
+                handleFailedResponse(response, responseString);
+                LOG.info(maskValueOfToken(responseString));
+                clusterId = mapper.readTree(responseString).get("id").asText();
+                waitForStateOfCluster(clusterId, "RUNNING", TimeUnit.MINUTES.toMillis(timeoutForClusterStateWait));
+                return getCloudCluster(clusterId);
+            }
+        } catch (Exception e) {
+            // Cluster id could be also empty if a problem occurred before cluster id exist
+            throw new CloudException(String.format("Create cluster with id %s is failed, Rc error is: %s", clusterId, getErrorString(e)));
+        }
+    }
+
+    private static String getErrorString(Exception e) {
+        return e + "\n" + Arrays.toString(e.getStackTrace());
+    }
+
+    private static void handleFailedResponse(Response response, String responseString) throws CloudException {
+        if (!response.isSuccessful()) {
+            String errorString = String.format("Unexpected http code %d Response: %s", response.code(), responseString);
+            LOG.error(errorString);
+            throw new CloudException(errorString);
+        }
+    }
+
+    public CloudCluster getCloudCluster(String clusterId) throws CloudException {
+        return getCloudCluster(clusterId, true);
+    }
+
+    public CloudCluster getCloudCluster(String clusterId, boolean setupTls) throws CloudException {
+        try {
+            try (Response res = sendGetRequest(String.format("/cluster/%s", clusterId))) {
+                ResponseBody responseBody = res.body();
+                if (responseBody == null) {
+                    throw new CloudException(String.format("Response body is null while getting a cluster: %s", res));
+                }
+                String responseString = responseBody.string();
+                try {
+                    handleFailedResponse(res, responseString);
+                } catch (CloudException e) {
+                    return null;
+                }
```

Comment:
Why do we have this exception-driven flow here? Why don't we throw the exception back to the caller?

Reply:
It is done like this for `waitForDeletedCluster`. I think we can add the try catch in waitForDeletedCluster. Let me see..